File: server/src/main/java/com/rsdp/entity/SchemeCandidate.java
package com.rsdp.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonRawValue;
import com.rsdp.config.typehandler.JsonbTypeHandler;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * AI 推荐候选清单实体。
 */
@Data
@TableName("scheme_candidate")
public class SchemeCandidate {

    @TableId
    private String candidateId;

    private String recommendRequestId;
    private String rspuId;
    private String rskuId;

    private BigDecimal score;
    private String aiReason;

    @JsonRawValue
    @TableField(typeHandler = JsonbTypeHandler.class)
    private String matchFactors;

    private String status;
    private String createdBy;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: web/src/types/schemeCandidate.ts
/**
 * AI 推荐候选状态（后端落库为小写）。
 */
export type SchemeCandidateStatus = 'pending' | 'accepted' | 'rejected'

/**
 * AI 推荐候选项（对应后端 SchemeCandidateResponse）。
 */
export interface SchemeCandidate {
  candidateId: string
  /** 推荐请求 ID（来源上下文） */
  recommendRequestId: string
  rspuId: string
  rspuName?: string
  primaryImageUrl?: string
  rskuId?: string
  /** AI 推荐得分 */
  score?: number
  /** AI 推荐理由 */
  aiReason?: string
  /** 匹配因子（JSONB，结构由 AI 侧决定） */
  matchFactors?: Record<string, unknown>
  status: string
  createdBy?: string
  createdAt?: string
  updatedAt?: string
}

/**
 * 创建 AI 推荐候选请求。
 */
export interface SchemeCandidateCreateRequest {
  recommendRequestId: string
  rspuId: string
  rskuId?: string
  score: number
  aiReason?: string
  matchFactors?: Record<string, unknown>
}

/**
 * 批量创建请求（candidates 逐项同创建请求）。
 */
export interface SchemeCandidateBatchCreateRequest {
  recommendRequestId: string
  candidates: SchemeCandidateCreateRequest[]
}

/**
 * 更新候选请求（接受/拒绝即 { status: 'accepted' | 'rejected' }）。
 */
export interface SchemeCandidateUpdateRequest {
  rskuId?: string
  score?: number
  aiReason?: string
  matchFactors?: Record<string, unknown>
  status?: SchemeCandidateStatus
}
